packages/vite-plugin-svelte/src/plugins/compile.js

```diff
@@ -32,37 +31,30 @@ export function compile(api) {
 				if (!svelteRequest || svelteRequest.raw) {
 					return;
 				}
-				const cache = api.getEnvironmentCache(this);
 				let compileData;
 				try {
-					const svelteMeta = this.getModuleInfo(id)?.meta?.svelte;
-					compileData = await compileSvelte(svelteRequest, code, options, svelteMeta?.preprocessed);
+					compileData = await compileSvelte(
+						svelteRequest,
+						code,
+						options,
+						this.getCombinedSourcemap()
```

Comment:
instead of passing in sourcemap from preprocessed that could be outdated because another transform changed the code in between the 2 plugins, use getCombinedSourceMap()  this has the added benefit that we don't have to pass preprocessed around in meta